### ISSUE TITLE
Allow customization of block preview height

### ISSUE THREAD

**Issue by jefferyto** [WordPress/gutenberg#60990]
## What problem does this address?

Before WordPress 6.5, the block preview height was not fixed, instead adapting to the height of the preview content. After 6.5, the block preview height appears to be fixed to 150px, I believe as a result of #59112.

150px is not tall enough for many (most?) blocks, but more importantly, any single fixed height will not work for all blocks.

## What is your proposed solution?

Support either one or (ideally) both:

* Allow the block developer to explicitly indicate the ideal preview height, similar to setting `viewportWidth`.
* Allow the preview to adapt to the height of the preview content.

**Comment by annezazu** [comment]
cc @t-hamano as you worked on that prior PR!

**Comment by t-hamano** [comment]
Thanks for the report.

#59112 is meant to fix bugs that were originally not working as expected. Additionally, the height applied to the preview is a minimum value rather than a fixed value.

Are there any steps to reproduce this problem?

**Comment by jefferyto** [comment]
> Additionally, the height applied to the preview is a minimum value rather than a fixed value.

It does not appear to be applied as a minimum value.

> Are there any steps to reproduce this problem?

1. Add a new post or page.
2. Open the block selector.
3. Hover over a block to see the preview.

I am using Firefox 125.0.2 on Ubuntu 23.10.

WordPress 6.4.2:

![6 4-1](https://github.com/WordPress/gutenberg/assets/252819/a0ae25ea-5ea6-4d41-9d2d-ceb6624beee4)
![6 4-2](https://github.com/WordPress/gutenberg/assets/252819/53fdcd87-d6ef-4f06-b21c-d285bc179d54)
![6 4-3](https://github.com/WordPress/gutenberg/assets/252819/2c58ae32-f645-4055-903c-35ed738dc065)
![6 4-4](https://github.com/WordPress/gutenberg/assets/252819/b3d8a295-4826-4034-a229-e618fda9ef13)
![6 4-5](https://github.com/WordPress/gutenberg/assets/252819/b969e991-a24b-4d2e-9178-890ad85cf073)

WordPress 6.5.2:

![6 5-1](https://github.com/WordPress/gutenberg/assets/252819/69c6ba3f-66df-41db-b8a5-a36c8b1616ad)
![6 5-2](https://github.com/WordPress/gutenberg/assets/252819/68286ca4-3c64-40f5-b118-74d4343bde69)
![6 5-3](https://github.com/WordPress/gutenberg/assets/252819/e33621e8-b965-4781-9d67-03954e24a677)
![6 5-4](https://github.com/WordPress/gutenberg/assets/252819/f64f58ea-067d-40cb-9a9d-fa454c6a1e76)
![6 5-5](https://github.com/WordPress/gutenberg/assets/252819/b5fb2b41-6e0f-4b69-b6ab-3580d698c824)

**Comment by t-hamano** [comment]
Thanks for the reply.

You seem to be testing on Ubuntu, but can you check if it reproduces on other OS or browsers?

I couldn't reproduce it in my environment. I hope others will test this issue as well.

### Environment Info

- OS: Windows 11
- Browser: Firefox 125.0.2 (64-bit)

![image](https://github.com/WordPress/gutenberg/assets/54422211/1de291ce-5c80-499d-91cc-94a7f4cdeadb)

**Comment by jefferyto** [comment]
I traced the issue to having this in my theme's editor stylesheet:

```css
body {
    height: 100%;
}
```

(More specifically, this was set for the front-end and included in the editor stylesheet.)

I assume the preview component is setting the iframe height based on the height of the `body` element. I suggest setting `height: auto` inline to override any theme CSS.

**Comment by t-hamano** [comment]
> I traced the issue to having this in my theme's editor stylesheet:

Could you tell me how this style is enqueued? If we can reproduce this problem in our environment, we may be able to decide whether to solve this problem with Gutenberg.

**Comment by jefferyto** [comment]
> Could you tell me how this style is enqueued?

My theme is a classic theme, the editor stylesheet is enqueued with:

```php
function after_setup_theme_editor_style() {
	add_editor_style();
}
add_action( 'after_setup_theme', 'after_setup_theme_editor_style' );
```

The stylesheets are written with Sass, and the editor stylesheet includes most of the styles from the front-end stylesheet as the theme also has some custom blocks.

**Comment by t-hamano** [comment]
I tested it with Twenty Twenty-One, but I can't reproduce it. (Regardless of whether Gutenberrg plugin is enabled)

I added `body { height: 100% }` [here](https://github.com/WordPress/wordpress-develop/blob/6974b994de5e9c707607dcb4233bd785865b4c6a/src/wp-content/themes/twentytwentyone/assets/css/style-editor.css#L2294-L2295).

As the screenshot shows, the body element inside the preview iframe has `height:auto` overriding `100%`.

![height-100](https://github.com/WordPress/gutenberg/assets/54422211/dd18ed33-4d59-4122-95bd-3f816ae6b90b)

**Comment by jefferyto** [comment]
For me the `body { height: 100%; }` in my editor stylesheet is transformed to `.editor-styles-wrapper { height: 100%; }` and so overrides `body { height: auto; }`. All other `body` references in the stylesheet are transformed to `.editor-styles-wrapper` when applied in the editor.

Edit: I made the same edit to Twenty Twenty-One and I see the issue (and `body` styles are still transformed to `.editor-styles-wrapper`).

**Comment by t-hamano** [comment]
I was able to reproduce this problem.

In order to reproduce this issue, the block editor must be _non-iframed_. That is, if you have a custom field enabled or a block registered with apiVersion 2.

However, this issue should not be caused by #59112. This is because #59112 makes changes to the `edit-site` package, but this issue can be reproduced using the post editor.

**Comment by t-hamano** [comment]
When I went back through the commit history, I found that this problem occurred in #57810.

I also provide detailed reproduction steps.

### Steps

- Enable Twenty Twenty-One and add `height: 100%;` here. This stylesheet is loaded into the editor via `add_editor_style()`.
- Enable custom fields in the editor. This will make the editor no longer an iframe.
- Open the main inserter, open the Patterns tab, and select one of the categories.

### ✅ Before #57810 is merged (commit hash: f6d29bd)

The `height:100%` defined in the theme is applied to the `body` element. This value will be overwritten by `auto`.

![before](https://github.com/WordPress/gutenberg/assets/54422211/ec96f3cd-0d01-4e74-9d5f-86f45898d491)

### ❌ Right after #57810 is merged (commit hash: 45eb9a5)

The `height:100%` defined in the theme is applied via the `.editor-styles-wrapper` selector. This value is not overwritten.

![after](https://github.com/WordPress/gutenberg/assets/54422211/c15ebbb3-8c3a-4a9b-a913-b6239493e069)
